injector/src/main/java/edu/ucr/cs/riple/injector/changes/RemoveTypeUseMarkerAnnotation.java

```diff
@@ -0,0 +1,100 @@
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package edu.ucr.cs.riple.injector.changes;
+
+import com.github.javaparser.Range;
+import com.github.javaparser.ast.expr.AnnotationExpr;
+import com.github.javaparser.ast.expr.MarkerAnnotationExpr;
+import com.github.javaparser.ast.nodeTypes.NodeWithAnnotations;
+import com.github.javaparser.ast.nodeTypes.NodeWithRange;
+import com.github.javaparser.ast.type.ClassOrInterfaceType;
+import com.github.javaparser.ast.type.Type;
+import edu.ucr.cs.riple.injector.Helper;
+import edu.ucr.cs.riple.injector.location.Location;
+import edu.ucr.cs.riple.injector.modifications.Deletion;
+import edu.ucr.cs.riple.injector.modifications.Modification;
+import edu.ucr.cs.riple.injector.modifications.MultiPositionModification;
+import java.util.HashSet;
+import java.util.Optional;
+import java.util.Set;
+import javax.annotation.Nullable;
+
+/**
+ * Removes a type-use marker annotation from a node in the source code. It will remove the
+ * annotation both from the declaration and all type arguments.
+ *
+ * <p>For instance, for the node: {@code java.util.@Nullable Map<@Nullable String,
+ * java.lang.@Nullable String> list;} It will remove the {@code @Nullable} annotation from the type
+ * {@code Map} and all the type arguments. The final output will be: {@code java.util.Map<String,
+ * java.lang.String> list;}
+ */
+public class RemoveTypeUseMarkerAnnotation extends RemoveMarkerAnnotation {
+
+  public RemoveTypeUseMarkerAnnotation(Location location, String annotation) {
+    super(location, annotation);
+  }
+
+  @Override
+  @Nullable
+  public <T extends NodeWithAnnotations<?> & NodeWithRange<?>>
+      Modification computeTextModificationOn(T node) {
+    Type type = Helper.getType(node);
+    AnnotationExpr annotationExpr = new MarkerAnnotationExpr(annotationName.simpleName);
+    Set<Modification> modifications = new HashSet<>();
+    // Remove the annotation from the declaration if exists e.g. @Annot String f;
+    Modification onDeclaration = super.computeTextModificationOn(node);
+    if (onDeclaration != null) {
+      modifications.add(onDeclaration);
+    } else {
+      // Remove the annotation from the type if exists e.g. java.lang.@Annot String f;
+      for (AnnotationExpr expr : type.getAnnotations()) {
+        if (expr.equals(annotationExpr)) {
+          Optional<Range> annotRange = expr.getRange();
+          annotRange
+              .map(value -> new Deletion(expr.toString(), value.begin, value.end))
+              .ifPresent(modifications::add);
+        }
+      }
+    }
+
+    // Remove annotation from type arguments if exists e.g. List<@Annot String>
```

Comment:
Is it possible to write something like `java.util.List<java.util.Map>` where the type argument is also fully qualified?  If so, do we insert and delete annotations from the right place for the type argument?  (should have a test for this case)

Reply:
Yes, we do have support for that, and have a test for deletion. For addition, I updated a test here: 24540c012d516b38bb30614fca77adb12ad0cdbb